rfcs/text/0000-alternative-planning.md

```diff
@@ -0,0 +1,315 @@
+then the DCP will run inefficiently or not at all. I'm concerned that the proposed process for creating these roadmaps and plans, as 
+described in the Roadmaps and Planning RFC and as an evolution of the current process, will not fit DCP organizational and development realities.
+
+This RFC tries to concisely describe those problems and propose alternative solutions. It comes as a result of being
+asked to write these alternatives up. Due to time constraints it's 
+nowhere near as comprehensive, well-written or referenced as I would like. Nonetheless, I hope it can help spur 
+constructive discussion.
+
+This RFC is the author's own point of view as a technical lead and someone with long experience in highly distributed 
+open-source projects that require co-operation between peer organizations and individuals.
+
+## Detailed Design
+
+From my understanding, the roadmaps and planning RFC describes a 70 day process where scientists, project leads, product owners, user experience
+experts and technical leads create a roadmap consisting of themes, each theme having one of more objectives. These 
+objectives are further described by user stories. Each objective is represented in the DCP Zenhub board as an epic with
+user stories represented by an item attached to the epic. Further items can be created as spikes to spend time clarifying
+the user stories and/or performing design work prior to implementation. Each item is assigned a milestone where there are
+3 milestones of equal length in a quarter.
+
+In my experience of the existing planning process almost all spikes are cross-component issues. DCP culture pretty much
+ mandates that these require an RFC. An RFC
+takes a minimum of 3 weeks from inception to approval, though realistically RFCs of any complexity require considerably
+more time to understand the problem and align with other planned and existing technical work.
+
+From my experience so far, if an RFC is required this means that it will take a minimum of 2 milestones (8 weeks)
+before it can be considered complete. A T-shirt estimate is only possible once the RFC is approved and there is a
+strong chance that estimate will be more than the remainder of the quarter (4 weeks). Which means it takes an entire
+quarter to plan work and you can only start it in the next quarter at the earliest. Even then, the roadmap process for
+that next quarter will have started long before so I fear that in the very worst case it could be *up to 9 months between planning and implementation*.
+In other words, we will have a waterfall of roadmapping, planning and implementation with an extremely long cycle time.
+
+![](../images/0000-process-cartoon.png)
+
+I have grave doubts about doing down this road for two reasons:
+
+* **Prediction is hard** - predicting what software you need to create is hard. It's harder the further out you try. 
+Surprisingly soon even the roughest prediction is impossible.
```

Comment:
That is a personal opinion rather than a fact. I've known many developers who can estimate and deliver on time and at quality.

Reply:
If that is a personal opinion then so is yours. Sure, it's easier to do such estimates in a predictable environment where all development is taking place in a single organization and site. But the point is that we have entirely the opposite here.

Reply:
I don't recall mentioning a _predictable_ environment. In fact, I was thinking about a situation where we were implementing WebSockets across Microsoft client and server while the JavaScript API (W3C) and the protocol (IETF) were in play at their respective standards organizations. Fun times.